src/pushprocessor.ts

```diff
@@ -189,7 +202,11 @@ function mergeRulesWithDefaults(
 
     let nextExpectedRuleIdIndex = 0;
     const newRules: IPushRule[] = [];
-    for (const rule of incomingRules.slice(0, firstCustomRuleIndex)) {
+    // Process the default rules by merging them with defaults
```

Comment:
merge the defaults with the defaults?

```suggestion
    // Merge our expected rules (including the incoming custom rules) into the
    // incoming default rules.